fastchat/llm_judge/gen_model_answer.py

```diff
@@ -106,24 +109,33 @@ def get_model_answers(
                 else:
                     do_sample = True
 
-                output_ids = model.generate(
-                    torch.as_tensor(input_ids).cuda(),
-                    do_sample=do_sample,
-                    temperature=temperature,
-                    max_new_tokens=max_new_token,
-                )
-                if model.config.is_encoder_decoder:
-                    output_ids = output_ids[0]
-                else:
-                    output_ids = output_ids[0][len(input_ids[0]) :]
-                output = tokenizer.decode(
-                    output_ids,
-                    skip_special_tokens=True,
-                    spaces_between_special_tokens=False,
-                )
-                if conv.stop_str:
-                    output = output[: output.find(conv.stop_str)]
-                output = output.strip()
+                # some models may error out when generating long outputs
+                try:
+                    output_ids = model.generate(
+                        torch.as_tensor(input_ids).cuda(),
+                        do_sample=do_sample,
+                        temperature=temperature,
+                        max_new_tokens=max_new_token,
+                    )
+                    if model.config.is_encoder_decoder:
+                        output_ids = output_ids[0]
+                    else:
+                        output_ids = output_ids[0][len(input_ids[0]) :]
+                    output = tokenizer.decode(
+                        output_ids,
+                        skip_special_tokens=True,
+                        spaces_between_special_tokens=False,
+                    )
+                    if conv.stop_str:
+                        output = output[: output.find(conv.stop_str)]
+                    output = output.strip()
+
+                    if conv.name == "xgen":
+                        output = output.replace("Assistant:", "").strip()
```

Comment:
Is something like `output = output[10:]` better?

Reply:
Not 100% sure the model will output "Assistant:" so I changed it to
```
if conv.name == "xgen" and output.startswith("Assistant:"):
    output = output.replace("Assistant:", "", 1).strip()
```
wdyt?